src/config-service.ts

```diff
@@ -0,0 +1,46 @@
+import path from 'path'
+import { cosmiconfig } from 'cosmiconfig'
+import { IConfigService, ISyncOptions, IDependencySection } from './types'
+import * as C from './cli-util'
+import { shrinkObject } from './util'
+
+const explorer = cosmiconfig('typesync')
+
+export function createConfigService(): IConfigService {
+  return {
+    readConfig: async (filePath: string) => {
+      const fileConfig: ISyncOptions = await explorer
+        .search(path.dirname(filePath))
+        .then((result) => (result ? result.config : {}))
+
+      const cliConfig = readCliConfig()
+
+      return { ...shrinkObject(fileConfig), ...shrinkObject(cliConfig) }
+    },
+  }
+}
+
+function readCliConfig(): ISyncOptions {
+  const { flags } = C.parseArguments(process.argv.slice(2))
```

Comment:
Can we pass in an `argv` object rather than depend on `process.argv`? Also makes testing easier.